pkg/apis/options/load.go

```diff
@@ -140,23 +141,25 @@ func isUnexported(name string) bool {
 
 // LoadYAML will load a YAML based configuration file into the options interface provided.
 func LoadYAML(configFileName string, into interface{}) error {
-	v := viper.New()
-	v.SetConfigFile(configFileName)
-	v.SetConfigType("yaml")
-	v.SetTypeByDefaultValue(true)
 
 	if configFileName == "" {
 		return errors.New("no configuration file provided")
 	}
 
-	data, err := os.ReadFile(configFileName)
+	unparsedBuffer, err := os.ReadFile(configFileName)
 	if err != nil {
 		return fmt.Errorf("unable to load config file: %w", err)
 	}
 
+	// We now parse over the yaml with env substring, and fill in the ENV's
+	buffer, err := envsubst.Bytes(unparsedBuffer)
+	if err != nil {
+		return fmt.Errorf("error in substituting env variables : %w", err)
+	}
+
 	// UnmarshalStrict will return an error if the config includes options that are
 	// not mapped to felds of the into struct
-	if err := yaml.UnmarshalStrict(data, into, yaml.DisallowUnknownFields); err != nil {
+	if err := yaml.UnmarshalStrict(buffer, into, yaml.DisallowUnknownFields); err != nil {
 		return fmt.Errorf("error unmarshalling config: %w", err)
```

Comment:
Our codeclimate rules only allow for 4 return statement per function.

Therefore, I would suggest moving the parsing of the envs and the yaml to a new internal function called either `parseYAML` or `parseConfiguration`

Reply:
Everything else looks good to me. Thank you for the good work.

Reply:
done.

We should really consider whether this code climate rule is valid (at least for functions with multiple "nil, err" returns ) as this change has actually made the code much less readable.

Reply:
Yes the rule seems a bit strict to me as well. A limit of 4 returns is quite low for a language like go.